rust/src/context.rs

```diff
@@ -18,11 +18,14 @@ impl Context {
     }
 
     /// Remove all registered plugins
-    pub fn reset(&mut self) {
```

Comment:
I don't think this mut was needed

Reply:
maybe it was there to symbolically indicate that the operation mutates an underlying resource, and if we only want that to happen when the resource is owned in the scope its mutated, then the `mut` keyword will prevent Rust from compiling anything that breaks this requirement. @zshipko?

Reply:
Yeah the intent was to prevent `reset` from being called when there are multiple references

Reply:
Okay, I'm just realizing this was not a drive by commit. Needing a mutable reference adds some complexity around the calling code. @zshipko Should I make the Context mutable and wrap in a mutex?